tests/client/oid.rs

```diff
@@ -0,0 +1,57 @@
+use mongodb::client::oid;
+use rustc_serialize::hex::ToHex;
+
```

Comment:
I want to see some test cases testing endianness. For example in the server, we have: https://github.com/mongodb/mongo/blob/master/src/mongo/bson/oid_test.cpp#L77